### PR TITLE
🌱 Improve release notes formatting

### PR DESCRIPTION
**Pull request by g-gaston**
**What this PR does / why we need it**:
A few life of quality improvements that automate manual steps performed to cleanup the release notes:
* Capitalize/shorten some area labels into more legible prefixes (API vs api, KCP vs provider/kubeadm-control-plane...)
* Remove backport PR prefixes "[release-x.x]"
* Omit "num of change of type x" lines when there are no changes of
  that kind (eg. no breaking changes)
* Use the area label for `documentation` to classify PRs that have the wrong emoji prefix in the title. We have seen in the past PR that use the 🐛 or 🌱 instead of 📖 , but they have the `area/documentation` label. In these cases, the label tends to be more reliable than the PR title, so this PR makes the label take precedence.

I put the last item in a separate commit, since we weren't completely sure if we wanted to automate this yet and if so, what should be the logic. Opening the conversation here.

Builds on top of #8817, so this needs to wait until that's merged. Review only the last two commits pls.

/area release

### CODE REVIEW

**Comment by g-gaston** [comment]
/hold
until https://github.com/kubernetes-sigs/cluster-api/pull/8817 merges

**Review by nawazkh — APPROVED** [review]
/lgtm
🚀

File: hack/tools/release/notes.go
Comment (fabriziopandini) [inline hack/tools/release/notes.go:74]: what about userFriendlyAreaLabels instead of areaLabelsFormatMapping

File: hack/tools/release/notes.go
Comment (fabriziopandini) [inline hack/tools/release/notes.go:167]: if userFriedlyLabel, exist := userFriendlyAreaLabels[area]; exist {

File: hack/tools/release/notes.go
Comment (fabriziopandini) [inline hack/tools/release/notes.go:294]: s/size/count

**Review by fabriziopandini — COMMENTED** [review]
few nits, otherwise lgtm for me

**Comment by fabriziopandini** [comment]
lgtm pending rebase on top of https://github.com/kubernetes-sigs/cluster-api/pull/8817

**Comment by g-gaston** [comment]
@fabriziopandini rebased 🙌🏻

**Comment by fabriziopandini** [comment]
/lgtm
@killianmuldoon @sbueringer PTAL when you have some time

**Review by killianmuldoon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**
